src/reconcile/rec_visibility.c

```diff
@@ -152,29 +152,26 @@ __rec_append_orig_value(
  *     Return if we need to save the update chain
  */
 static bool
-__rec_need_save_upd(WT_SESSION_IMPL *session, WT_UPDATE *selected_upd, uint64_t max_txn,
-  wt_timestamp_t max_ts, bool has_newer_updates, uint64_t flags, WT_PAGE *page)
+__rec_need_save_upd(
+  WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_UPDATE_SELECT *upd_select, bool has_newer_updates)
 {
+    if (F_ISSET(r, WT_REC_EVICT) && has_newer_updates)
```

Comment:
There is a case in `evict_file` where we don't set the `WT_REC_HS` flag for eviction. In the previous versions, this would fire an assertion complaining we shouldn't have saved updates when we discard the page. So I have to put the section:
```
if (!LF_ISSET(WT_REC_HS))	
        return false;
```
before
```
if (LF_ISSET(WT_REC_EVICT) && has_newer_updates)	
        return true;
```
to handle that.
I think something have changed and it is not firing anymore, which is good.